services/app-api/utils/time/time.ts

```diff
@@ -21,18 +21,80 @@ export const convertDateUtcToEt = (date: number): string => {
 };
 
 /*
- * Calculates the period given a due date.
+ * Calculates the period given the current date.
  * The periods are defined as follows:
  *     Period 1 is from 01/01 to 06/30.
  *     Period 2 is from 07/01 to 12/31.
  */
-export const calculatePeriod = (dueDate: number, workPlan?: ReportMetadata) => {
+export const calculatePeriod = (
+  currentDate: number,
```

Comment:
currentDate seems like a funny thing to pass in when you could just use Date.now() 🤔 
I know we get Date.now() in create.ts so we can pass it in easily, just thinking out loud. Does it hurt to calculate it here as well?

Reply:
There also seems to be some funny logic in general around dates and years sometimes be calculated and sometimes coming from metadata. Perhaps a larger thing to think about (or totally ignore 👍)

Reply:
> There also seems to be some funny logic in general around dates and years sometimes be calculated and sometimes coming from metadata. Perhaps a larger thing to think about (or totally ignore 👍)

ohhh hmmm very good point.... I will have to think about this one some more

> currentDate seems like a funny thing to pass in when you could just use Date.now() 🤔
I know we get Date.now() in create.ts so we can pass it in easily, just thinking out loud. Does it hurt to calculate it here as well?

true. we could just calculate `Date.now()` here. also it looks like we only need the month here? so maybe we could pass in the month from create.ts? lots of paths here 🤔

Reply:
On second thought, I like having the date as a parameter! it makes testing easier 😄 we can pass in a variety of dates and verify that the period is what we expect it to be. That becomes a bit trickier to do if `calculatePeriod` computes `Date.now()` itself

Reply:
true enough!